src/sparseml/transformers/text_classification.py

```diff
@@ -674,19 +667,26 @@ def _create_train_dataset(raw_datasets, max_train_samples):
                 )
             eval_dataset = raw_datasets["validation"]
         elif data_args.validation_ratio is not None:
-            if train_dataset is None:
-                train_dataset = _create_train_dataset(
-                    raw_datasets, data_args.max_train_samples
-                )
+            train_dataset = (
+                raw_datasets["train"] if train_dataset is None else train_dataset
+            )
             train_dataset, eval_dataset = _split_train_val(
                 train_dataset, data_args.validation_ratio
             )
+            if (
+                data_args.max_train_samples is not None
+                and len(train_dataset) > data_args.max_train_samples
```

Comment:
Should we also support cases where `len(train_dataset) < data_args.max_train_samples` ?

Reply:
Or would that qualify as a usage error

Reply:
No (if max_train_samples >= len of the dataset then we just keep using it)